synapseclient/core/async_utils.py

```diff
@@ -48,3 +51,73 @@ async def otel_trace_method_wrapper(self, *arg, **kwargs) -> None:
         return otel_trace_method_wrapper
 
     return decorator
+
+
+class ClassOrInstance:
+    """Helper class to allow a method to be called as a class method or instance method."""
+
+    def __init__(self, fn):
+        self.fn = fn
+
+    def __get__(self, obj, cls):
+        def f(*args, **kwds):
+            if obj is not None:
+                return self.fn(obj, *args, **kwds)
+            else:
+                return self.fn(cls, *args, **kwds)
+
+        functools.update_wrapper(f, self.fn)
+        return f
+
+
+# Adapted from https://github.com/keflavich/astroquery/blob/30deafc3aa057916bcdca70733cba748f1b36b64/astroquery/utils/process_asyncs.py#L11
+def async_to_sync(cls):
```

Comment:
The is the defining change that allows us to create the wrappers for the async functions at run-time.